std/file.d

```diff
@@ -3142,7 +3142,7 @@ private struct DirIteratorImpl
 
         bool stepIn(string directory)
         {
-            auto h = cenforce(opendir(directory.tempCString()), directory);
+            auto h = cenforce(opendir(directory.length ? directory.tempCString() : "."), directory);
```

Comment:
```d
auto cdir = directory.length ? directory.tempCString() : "."; // cdir may be invalid after this line
```

It is not possible to conditionally initialize a `tempCString` variable because it returns a voldemort type with `@disable this()`.

Alternative is:
```d
tempCString(directory.length ? directory : ".");
```
but this initializes "." to a C string redundantly.